ticket_upgrade_prediction/evaluator.py

```diff
@@ -39,8 +41,20 @@ def to_dict(self) -> dict:
         return self.__dict__
 
 
-class Evaluator:
-    def __init__(self, model: Any, X: pd.DataFrame, y: np.array) -> None:
+class BaseEvaluator(ABC):
+    """Interface for Evaluator"""
+
+    @abstractmethod
+    def get_all_metrics(self, to_mlflow: bool):
+        """Method for getting all metrics"""
+
+    @abstractmethod
+    def plot_all_plots(self, save_path: Path, to_mlflow: bool):
+        """Methods for plotting everything"""
+
+
```

Comment:
Hmm, on first tought, i felt like it should be here, since there will be only one evaluator. Yet, if we consider AutoML for future WEC, it mind be useful for regression evaluator